base/some.jl

```diff
@@ -95,9 +95,9 @@ ERROR: ArgumentError: No value arguments present
 function something end
 
 something() = throw(ArgumentError("No value arguments present"))
-something(x::Nothing, y...) = something(y...)
-something(x::Some, y...) = x.value
-something(x::Any, y...) = x
+something(x::Nothing, @nospecialize(y...)) = something(y...)
```

Comment:
This one seems to be sketchy, as specialization may produce better code for the `y...` part. We should double check the performance for splat cases.

Reply:
You way first understand what `@nospecialize` really does. Read https://github.com/JuliaLang/julia/pull/41931 if interested.